container/common/fsHandler.go

```diff
@@ -119,16 +149,42 @@ func (fh *realFsHandler) trackUsage() {
 	}
 }
 
+func (fh *realFsHandler) trackInodeUsage() {
+	fh.updateInodeUsage()
+	for {
+		select {
+		case <-fh.stopChan:
+			return
+		case <-time.After(fh.findPeriod):
+			start := time.Now()
+			if err := fh.updateInodeUsage(); err != nil {
+				glog.Errorf("failed to collect filesystem stats using Find - %v", err)
+				fh.findPeriod = fh.findPeriod * 2
+				if fh.findPeriod > maxFindBackoffFactor*fh.minFindPeriod {
+					fh.findPeriod = maxFindBackoffFactor * fh.minFindPeriod
+				}
+			} else {
+				fh.findPeriod = fh.minFindPeriod
+			}
+			duration := time.Since(start)
+			if duration > longFind {
+				glog.V(2).Infof("`find` on following dirs took %v: %v", duration, []string{fh.rootfs, fh.extraDir})
+			}
+		}
+	}
+}
```

Comment:
Actually, on further thought I'm wondering if this should be in the same routine as the `du` call. The advantages are that containers with very large filesystems wouldn't hog as many exec calls, and there might be less disk thrashing.